### PR TITLE
fail manpage-generation when tool fails on --version --help/--help-all

### PR DESCRIPTION
**Pull request by wenningerk**

### CODE REVIEW

File: tools/crm_attribute.c [hunk 1]
```diff
@@ -326,7 +326,7 @@ main(int argc, char **argv)
         g_strfreev(processed_args);
         pcmk__free_arg_context(context);
         /* FIXME:  When crm_attribute is converted to use formatted output, this can go. */
-        pcmk__cli_help('v', CRM_EX_USAGE);
+        pcmk__cli_help('v', CRM_EX_OK);
```
Comment (kgaillot) [inline tools/crm_attribute.c:329]: This is a regression in 2.1.0 :-/

File: tools/crm_diff.c [hunk 2]
```diff
@@ -322,7 +322,7 @@ main(int argc, char **argv)
         g_strfreev(processed_args);
         pcmk__free_arg_context(context);
         /* FIXME:  When crm_diff is converted to use formatted output, this can go. */
-        pcmk__cli_help('v', CRM_EX_USAGE);
+        pcmk__cli_help('v', CRM_EX_OK);
```
Comment (kgaillot) [inline tools/crm_diff.c:325]: regression in 2.0.4

File: tools/crm_error.c [hunk 3]
```diff
@@ -93,7 +93,7 @@ main(int argc, char **argv)
         g_strfreev(processed_args);
         pcmk__free_arg_context(context);
         /* FIXME:  When crm_error is converted to use formatted output, this can go. */
-        pcmk__cli_help('v', CRM_EX_USAGE);
+        pcmk__cli_help('v', CRM_EX_OK);
```
Comment (kgaillot) [inline tools/crm_error.c:96]: regression in 2.0.4

File: tools/crm_node.c [hunk 4]
```diff
@@ -549,7 +549,7 @@ main(int argc, char **argv)
         g_strfreev(processed_args);
         pcmk__free_arg_context(context);
         /* FIXME:  When crm_node is converted to use formatted output, this can go. */
-        pcmk__cli_help('v', CRM_EX_USAGE);
+        pcmk__cli_help('v', CRM_EX_OK);
```
Comment (kgaillot) [inline tools/crm_node.c:552]: regression in 2.0.3

File: tools/crm_rule.c [hunk 5]
```diff
@@ -252,7 +252,7 @@ main(int argc, char **argv)
         g_strfreev(processed_args);
         pcmk__free_arg_context(context);
         /* FIXME:  When crm_rule is converted to use formatted output, this can go. */
-        pcmk__cli_help('v', CRM_EX_USAGE);
+        pcmk__cli_help('v', CRM_EX_OK);
```
Comment (kgaillot) [inline tools/crm_rule.c:255]: regression in 2.0.4

**Review by kgaillot — COMMENTED** [review]
Looks good. I should have merged this before releasing rc3, but oh well ...

**Comment by wenningerk** [comment]
Yeah though it might be interesting for rc3 - which is why I had written a heads-up on irc.
But you were quite far with rc3 then already. Simple as this thing is ... things like --help vs. --help-all
and stuff took me a bit to understand.

**Comment by kgaillot** [comment]
> Yeah though it might be interesting for rc3 - which is why I had written a heads-up on irc.
> But you were quite far with rc3 then already. Simple as this thing is ... things like --help vs. --help-all
> and stuff took me a bit to understand.

Yeah I've never liked --help-all. There's a glib option to make --help behave like --help-all, maybe we should turn that on. Some of our tools do have a zillion options, though, so I can see why some people might like the condensed --help.

**Comment by wenningerk** [comment]
> > Yeah though it might be interesting for rc3 - which is why I had written a heads-up on irc.
> > But you were quite far with rc3 then already. Simple as this thing is ... things like --help vs. --help-all
> > and stuff took me a bit to understand.
> 
> Yeah I've never liked --help-all. There's a glib option to make --help behave like --help-all, maybe we should turn that on. Some of our tools do have a zillion options, though, so I can see why some people might like the condensed --help.

Didn't say I was against --help-all btw. just had to learn when to try what.
Would be interesting if in the end we couldn't have just taken back --no-discard-stderr.
Anyway this thing seems to be working and we don't have to check if there is a tool that prints to stderr.
And poor documentation on the help2man-side as the man page definitely doesn't mention that
--no-discard-stderr on top of merging stdout & stderr as well discards the exit-codes. Should probably
be switchable separately both for merging & exit-code handling and for --version & --help.
In case of --version, merging + discarding the exit-code is very likely a bad idea anyway.

**Pull request merged**